dask/array/tests/test_overlap.py

```diff
@@ -265,6 +265,16 @@ def test_overlap():
     assert g.chunks == ((8, 8), (5, 5))
 
 
+def test_overlap_allow_rechunk_kwarg():
```

Comment:
These tests could go with the original tests of map_overlap, since the same results should be gotten when the smallest chunk size is greater than depth.  We could make that test parameterized and run it with both allow_rechunk=True and allow_rechunk=False.

The tests that have the smallest chunk less than depth can be either pulled out, or checked that they raise when allow_rechunk=False.